lms/djangoapps/django_comment_client/utils.py

```diff
@@ -137,6 +138,29 @@ def get_accessible_discussion_modules(course, user, include_all=False):  # pylin
     ]
 
 
+def get_discussion_module(course, user, discussion_id):
```

Comment:
@robrap For discussion modules of a course; category is the module type `e.g. 'Getting Started', 'Google Hangout', 'Getting Graded', 'Essays' etc.`

Each module category type could have multiple discussion targets e.g. 'Getting Started' have 'Working with Videos' where the discussion exists. At that level it is courseware discussion and the ID of that discussion is used to identify it and also referred as `topic_id` for that discussion. The name 'Working with Videos' also appears in topic dropdown selection in `forums` tab.

If 'General' topic type is selected to create a post from forums tab; we consider it as non-courseware topic.

Now; `platform` and `cs_comments_service` consider both `topic_id` and `discussion_id` as `commentable_id`.
For `a discussion id (a.k.a topic id); would that be 100% accurate always` I believe @nasthagiri could give a better insight on this.